gprofiler/metadata/application_identifiers.py

```diff
@@ -317,26 +354,36 @@ def get_app_id(process: Process, identifiers: List[_ApplicationIdentifier]) -> O
     if not _ApplicationIdentifier.enrichment_options.application_identifiers:
         return None
 
+    appids = []
     for identifier in identifiers:
         try:
             appid = identifier.get_app_id(process)
             if appid is not None:
-                return appid
+                if not aggregate_all:
+                    return appid
+                appids.append(appid)
 
         except NoSuchProcess:
+            if aggregate_all and len(appids) != 0:
+                return ", ".join(appids)
             return None
         except Exception:
             _logger.exception(
                 f"Application identifier {identifier} raised an exception while matching against process {process}"
             )
             continue
 
+    if aggregate_all and len(appids) != 0:
+        return ", ".join(appids)
```

Comment:
```suggestion
    if len(appids) != 0:
        return ", ".join(appids)
```
you cannot reach here otherwise